src/auth/oauth2client.ts

```diff
@@ -130,13 +223,13 @@ export class OAuth2Client extends AuthClient {
    * The base URL for auth endpoints.
    */
   private static readonly GOOGLE_OAUTH2_AUTH_BASE_URL_ =
-      'https://accounts.google.com/o/oauth2/auth';
+      'https://accounts.google.com/o/oauth2/v2/auth';
 
   /**
    * The base endpoint for token retrieval.
    */
   private static readonly GOOGLE_OAUTH2_TOKEN_URL_ =
-      'https://accounts.google.com/o/oauth2/token';
+      'https://www.googleapis.com/oauth2/v4/token';
```

Comment:
What is the semver impact of this change?

Reply:
Also: the justification for this change needs to be documented in the PR and the commit message.

Reply:
I would say this is a semver minor change.  I have no reason to believe it breaks any functionality, but it does (I think) bring us up to date with the most recent documentation:
https://developers.google.com/identity/protocols/OAuth2InstalledApp

It's also suggested in #120 that this is causing problems for users as well.  I will update the PR and commit message to call these both out.